src/components/AuthDialog/SignUp/SignUp.tsx

```diff
@@ -0,0 +1,117 @@
+'use client'
+
+import { yupResolver } from '@hookform/resolvers/yup'
+import { SubmitHandler, useForm } from 'react-hook-form'
+
+import { Button } from '@/components/ui/button'
+import {
+  Card,
+  CardContent,
+  CardDescription,
+  CardFooter,
+  CardHeader,
+  CardTitle,
+} from '@/components/ui/card'
+import { Input } from '@/components/ui/input'
+import { signUpSchema } from '@/schemas/authSchemas'
+import { SignUpForm } from '@/types/auth'
+
+interface AuthDialogProps {
+  switchToLogin: () => void
+}
+
+const SignUp: React.FC<AuthDialogProps> = ({ switchToLogin }) => {
+  const signUpMethods = useForm<SignUpForm>({
+    resolver: yupResolver(signUpSchema as any),
```

Comment:
You can remove the any here

Reply:
The confirmPassword is optional in this validation schema. I just need to mark it as optional in the form to handle the error, instead of considering signUpSchema as any [74810f1](https://github.com/dark-orga/anitube-website/pull/2/commits/74810f1ca9ded7c30bb6f4f2e3fb8e9be2f46431)